src/app/features/post/post.component.ts

```diff
@@ -113,9 +116,23 @@ export class PostComponent {
     private router: Router,
     private markdownService: MarkdownService,
     @Inject(DOCUMENT) private document: Document,
-    private htmlInMarkdownService: HtmlInMarkdownService
+    private htmlInMarkdownService: HtmlInMarkdownService,
+    private postUrlPipe: PostUrlPipe,
+    private meta: Meta
   ) {}
 
+  ngOnInit(): void {
+    this.post$
+      .pipe(
+        filter((value): value is Post => !!value),
+        map(value => this.postUrlPipe.transform(value, value.teaser)),
+        first()
+      )
+      .subscribe(content => {
+        this.meta.updateTag({ name: 'thumbnail', content });
```

Comment:
We should use open graph protocol to have it properly indexed, so the name must be `og:image`. It should also be the absolute URL.

Since this is used for indexing, it might be a good idea to include `og:url` as well with the post URL.

Reply:
Also, please check if this implementation works in the scenario where I navigate to another post from the current one – like clicking on a related post at the bottom. Since you have an observable getting the first emission in ngOnInit, it might not subscribe another time after PostComponent is activated.

My suggestion would be tapping the observable and rely on the async subscription

Reply:
Could you check my las changes?
I hope I did what you expected🙂